fairscale/nn/data_parallel/sharded_ddp.py

```diff
@@ -367,33 +366,15 @@ def _clear_counters(self) -> None:
         self._grad_to_be_reduced = [True for _ in self._grad_to_be_reduced]
         self._bucket_flush_callback_set = False
 
-        # Do not reset the buckets
         if self.use_buckets:
             assert self._bucket_list is not None
 
             for bucket in self._bucket_list:
-                assert (
```

Comment:
this was useful prior to the .Variable callback method, not useful anymore since the buckets will always be flushed if there has been a BW. The attached issue was tied to this and a re-entry (train, eval, train again). This assert would trigger on the second train, since the previous step was a FW without a BW pass, it was not doing a very good job in that case (no grads in this bucket really, it was just the .sent flag which did not flip).

Reply:
see l413 for the bucket flush callback